wireless_modules/middle_module/main.py

```diff
@@ -51,35 +49,43 @@ def sub_cb(topic, msg):
 
 
 def restart_and_reconnect():
-    print('Failed to connect to MQTT broker. Reconnecting...')
+    print("Failed to connect to MQTT broker. Reconnecting...")
     utime.sleep(10)
     machine.reset()
 
 
-# Create Client class instance
-my_client = Client(client_id, config.MQTT_BROKER)
+def publish_test_messages(client, topic):
+    """
+    Keeps publishing data till a stop message is received
+    :param client: A instance of the Client class
+    :param topic: Topic to publish on
+    """
+    test_no = 0
+    while start_publish:
+        message = "Message sample " + str(test_no)
+        client.mqtt_pub(message, topic)
+        print("MQTT data sent: %s on %s through %s" % (message, PUB_DATA_TOPIC, config.MQTT_BROKER))
+
+        client.check_for_message()
 
-if not my_client.connect_and_subscribe(SUB_TOPICS, sub_cb):
-    # If the connect_and_subscribe returns False, a connection to the broker could not be established
-    restart_and_reconnect()
+        utime.sleep(1)
+        test_no = test_no + 1
```

Comment:
Stating this for other reviewers to see - this will only start/stop once before exiting which is a known issue, but since this PR is mainly for the MQTT client class and the rest is WIP we'll iron out stuff like this later.

Reply:
perhaps add a `FIXME:` comment here?

Reply:
So I can make it continuously check for the start message again (The reason why it didn't work before was because of ampy (The thing that let me run scripts on ESP32) and so I have started using picocom instead which works with infinite loops). But I'll probably make the change in the next PR since this PR was mostly to get the MQTT class file out there